packages/offix-datastore/src/models.ts

```diff
@@ -1,21 +1,41 @@
-import { GraphQLSchema, GraphQLObjectType } from "graphql";
-import { parseMarker } from "graphql-metadata";
-import { getUserTypesFromSchema } from "@graphql-toolkit/common";
+import { Storage, StoreChangeEvent } from "./storage";
+import { createPredicate } from "./predicates";
 
-export interface Model {
-    __typename: string;
-}
+export type EventTypes = "ADD" | "UPDATE" | "DELETE";
 
-export interface PersistedModel extends Model {
-    id: string;
-}
+export class Model<T> {
+    constructor(
+        private storeName: string,
+        private schema: any,
```

Comment:
Seems unused. What is the purpose of the schema

Reply:
The schema is the fields you referred to earlier. I'll rename the variables appropriately. I use it to generate ModelPredicate. Take a look at the `query` function